File: server/src/domain/gym-challenges/dto/UpdateGymChallengeDto.ts
import { z } from "zod";

export const UpdateGymChallengeSchema = z.object({
  id: z.number().int(),
  name: z.string().optional(),
  description: z.string().optional(),
  image: z.string().url().optional(),
});

export type UpdateGymChallengeDto = z.infer<typeof UpdateGymChallengeSchema>;

File: jim_hats_mobile/lib/data/gym_challenges/dtos/update_gym_challenge_dto.dart

class UpdateGymChallengeDto {
  final String? name;
  final String? description;
  // DD/MM/YYYY date
  final String? startAt;
  // DD/MM/YYYY date
  final String? endAt;
  final String? image;

  UpdateGymChallengeDto({
    this.name,
    this.description,
    this.startAt,
    this.endAt,
    this.image,
  });

  Map<String, dynamic> toMap() {
    return <String, dynamic>{
      'name': name,
      'description': description,
      'startAt': startAt,
      'endAt': endAt,
      'image': image,
    };
  }


}
